File: nebulous/ems-core/broker-client/samples/sample.js
var BrokerClient = Packages.gr.iccs.imu.ems.brokerclient.BrokerClient;
var System = Packages.java.lang.System;

var sleep = Packages.java.lang.Thread.sleep;
var currentTimeMillis = Packages.java.lang.System.currentTimeMillis;
var newClient = Packages.gr.iccs.imu.ems.brokerclient.BrokerClient.newClient;
var getMessageListener = Packages.gr.iccs.imu.ems.brokerclient.BrokerClientApp.getMessageListener;

print("Time: "+currentTimeMillis());
print("Args: "+args);

var topic = argsArray[0];
var value = parseFloat(argsArray[1]);
var url = argsArray.length> 2 ? argsArray[2] : 'tcp://localhost:61616';
var username = argsArray.length > 3 ? argsArray[3] : '';
var password = argsArray.length > 4 ? argsArray[4] : '';

main();

function main() {
    var bc = newClient(username, password);
    bc.getClientProperties().setPreserveConnection(true);
    //print("BC: "+bc.getClientProperties());

    list_topics(bc, url);

    print(`Subscribing for events to topic ${topic}...`);
    var listener = getMessageListener('', '');
    var exit_callback = null;
    bc.subscribe(url, topic, listener);

    print("Publishing events...");
    bc.publishEvent(url, topic, { "metricValue": value, "level": 1, "timestamp": currentTimeMillis() }, { 'AAA': 'aaa', 'BBB': 'bbb' });
    bc.publishEvent(url, topic, 'MAP', '{ "command": "do something" }', { 'AAA': 'aaa', 'BBB': 'bbb' });

    sleep(2 * 1000);
    list_topics(bc, url);

    bc.unsubscribe(listener);
    bc.closeConnection();
}

function list_topics(bc, url) {
    print();
    print('Topics at: ' + url);
    for (s of bc.getDestinationNamesSorted(url)) {
        print('  '+s);
    }
    print();
}

File: nebulous/ems-core/broker-client/samples/sample.py
import time
from java.lang import Object
from java.lang import System
from gr.iccs.imu.ems.brokerclient import BrokerClient
from gr.iccs.imu.ems.brokerclient import BrokerClientApp
from gr.iccs.imu.ems.brokerclient.event import EventMap
from gr.iccs.imu.ems.brokerclient.BrokerClient import MESSAGE_TYPE
from gr.iccs.imu.ems.brokerclient.BrokerClient import ON_EXCEPTION

topic = args[0]
value = float(args[1])
url = args[2] if len(args) > 2 else 'tcp://localhost:61616'
username = args[3] if len(args) > 3 else ''
password = args[4] if len(args) > 4 else ''

bc = BrokerClient.newClient(username, password)
bc.getClientProperties().setPreserveConnection(1)

def main():
    list_topics()

    print "Subscribing for events to topic %s..." % topic
    # listener = lambda message: System.out.println("Received: %s - %d" % message.getClass().getSimpleName(), message.getBody(Object.class))
    listener = BrokerClientApp.getMessageListener('', '')
    exit_callback = lambda exit_code: None
    bc.subscribe(url, topic, listener)
    # bc.subscribeWithAutoReconnect(url, topic, listener, ON_EXCEPTION.LOG_AND_IGNORE, exit_callback)

    print "Publishing events..."
    bc.publishEvent(url, topic, MESSAGE_TYPE.TEXT, EventMap(value, 1, System.currentTimeMillis()), {'PPPP': '1111', 'RRRRR': '22222'})
    bc.publishEvent(url, topic, 'MAP', '{ "command": "do something" }', { 'AAA': 'aaa', 'BBB': 'bbb' })
    # BrokerClientApp.sendEvent(url, username, password, topic, 'MAP', '{ "command": "do something" }', { 'AAA': 'aaa', 'BBB': 'bbb' })

    time.sleep(2)
    list_topics()

    bc.unsubscribe(listener)
    bc.closeConnection()


def list_topics():
    print
    print "Topics at: ", url
    for s in sorted(bc.getDestinationNames(url)):
        print(s)
    print

main()
